### PR TITLE
Add model ready waiting logic to resolve issue #331

### PR DESCRIPTION
**Pull request by jim-wang-intel**
- Add gRPC calls in profile-launcher to test models readiness from ovms server
 - Refactor how the unit-test makefile runs as now it requires google gRPC packages

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [x] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## Issue this PR will close

close: #**331**

## Test Instructions if applicable

- git clone this PR
- make build-profile-launcher
- make clean-all
- make run-camera-simulator
- run multiple pipelines, say, classification, for more than 5 pipelines and observe that the 6th nd 7th pipelines can also be launched without problem now:
```console
$ PIPELINE_PROFILE="classification" RENDER_MODE=0 sudo -E ./run.sh --workload ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0

```
- run unit tests:  `make unit-test-profile-launcher`  should see ALL PASSED ok
- run smoke tests: `make run-smoke-tests` should see ALL PASSED

### CODE REVIEW

File: configs/opencv-ovms/cmd_client/main.go [hunk 1]
```diff
@@ -29,9 +30,14 @@ import (
 	"path/filepath"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
+	"github.com/intel-retail/vision-self-checkout/configs/opencv-ovms/cmd_client/parser"
 	"github.com/intel-retail/vision-self-checkout/configs/opencv-ovms/cmd_client/portfinder"
+
+	grpc_client "github.com/intel-retail/vision-self-checkout/configs/opencv-ovms/cmd_client/grpc-client"
```
Comment (antoniomtz) [inline configs/opencv-ovms/cmd_client/main.go:39]: should this URL be `github.com/intel-retail/automated-self-checkout` ?

**Review by antoniomtz — CHANGES_REQUESTED** [review]

**Review by vli11 — APPROVED** [review]
run and unit test are good

File: configs/opencv-ovms/cmd_client/main.go (hunk 1 above)
Comment (jim-wang-intel) [reply configs/opencv-ovms/cmd_client/main.go:39]: it is related to go.mod module definition: i can change all the vision-self-checkout to automated-self-check in the current code including the go.mod.

Comment (jim-wang-intel) [reply configs/opencv-ovms/cmd_client/main.go]: @antoniomtz fixed.

**Review by antoniomtz — APPROVED** [review]
LGTM

File: configs/opencv-ovms/cmd_client/main.go [hunk 2]
```diff
@@ -342,7 +352,110 @@ func (ovmsClientConf *OvmsClientConfig) startOvmsServer() {
...
+		// sleep a bit and retry it again
+		time.Sleep(time.Second)
+	}
+
+	log.Println("OVMS server started and ready to serve")
+}
+
+func (ovmsClientConf *OvmsClientConfig) waitForOvmsModelsReady() error {
+	grpcPort := os.Getenv(GRPC_PORT_ENV)
+	ovmsURL := fmt.Sprintf("%s:%s", "localhost", grpcPort)
```
Comment (brian-intel) [inline configs/opencv-ovms/cmd_client/main.go:383]: Moving the "localhost" to a variable will enable different hosts. The main case being a remote OVMS server.

Comment (jim-wang-intel) [reply configs/opencv-ovms/cmd_client/main.go:383]: @brian-intel    let's deal with this in another enhancement PR as of today we are running docker in host mode network so it won't be able to support the remote yet

**Pull request merged**